pkg/apis/redhatcop/v1alpha1/namespaceconfig_types.go

```diff
@@ -1,42 +1,54 @@
 package v1alpha1
 
 import (
+	"github.com/redhat-cop/operator-utils/pkg/util/apis"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/runtime"
 )
 
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
 // NamespaceConfigSpec defines the desired state of NamespaceConfig
-// +k8s:openapi-gen=true
+// There are two selectors: "labelSelector", "annotationSelector".
+// Selectors are considered in AND, so if multiple are defined they must all be true for a Namespace to be selected.
 type NamespaceConfigSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
-	// Add custom validation using kubebuilder tags: https://book.kubebuilder.io/beyond_basics/generating_crd.html
+	// Add custom validation using kubebuilder tags: https://book-v1.book.kubebuilder.io/beyond_basics/generating_crd.html
 
-	Selector  metav1.LabelSelector   `json:"selector,omitempty"`
-	Resources []runtime.RawExtension `json:"resources,omitempty"`
+	// LabelSelector selects Namespaces by label.
+	// +kubebuilder:validation:Required
+	LabelSelector metav1.LabelSelector `json:"labelSelector"`
+
+	// AnnotationSelector selects Namespaces by annotation.
+	// +kubebuilder:validation:Required
```

Comment:
The examples are outdated due to the changed selector:
```
The NamespaceConfig "overcommit-limitrange" is invalid: 
* spec.annotationSelector: Required value
* spec.labelSelector: Required value
```

Do both selectors have to be required though? It feels a bit redundant to add `annotationSelector: {}` to everything if you only want to use the labelSelector.

```
apiVersion: redhatcop.redhat.io/v1alpha1
kind: NamespaceConfig
metadata:
  name: overcommit-limitrange
spec:
  labelSelector:
    matchLabels:
      overcommit: "limited"
  annotationSelector: {}
  resources:    
  - apiVersion: "v1"
    kind: "LimitRange"
    metadata:
      name: "overcommit-limits" 
    spec:
      limits:
        - type: "Container" 
          maxLimitRequestRatio:
            cpu: 100
            memory: 1
```

Reply:
should have been optional, fixed